### PR TITLE
Update Minimum MySQL requirements and also update to indicate that Ci…

### PR DESCRIPTION
**Pull request by seamuslee001**
…viCRM 5.24 and above can run on MySQL 8 but not yet recommended

ping @eileenmcnaughton @totten

### CODE REVIEW

File: docs/requirements.md [hunk 1]
```diff
@@ -105,11 +105,11 @@ Other database servers (such as PostgreSQL) are not compatible with CiviCRM.
 
 ### MySQL Version
 
-Your MySQL version should be **5.7.5 or greater** or MariaDB **10.0.2 or greater**.  CiviCRM does still work on older versions but there is some functionality missing, notably around ensuring various database actions don't compete for the same resources.
+Your MySQL version should be **5.7.5 or greater** or MariaDB **10.0.2 or greater**.  As of version 5.26 the minimum install version for CiviCRM is 5.5, users on versions before that are advised to upgrade their MySQL instance to a recommended version.
```
Comment (eileenmcnaughton) [inline docs/requirements.md:108]: We should still tell 5.5 & 5.6 users they will be missing some functionality

File: docs/requirements.md [hunk 2]
```diff
@@ -105,11 +105,11 @@ Other database servers (such as PostgreSQL) are not compatible with CiviCRM.
...
+CiviCRM versions as of 5.24 has been shown to be able to run on MySQL 8 through the execution of our test matrix. It is not yet recommended as not all of the Content Management Systems support MySQL 8, CiviCRM MySQL 8 support is being tracked in the following issue: [open issue for MySQL 8 support](https://lab.civicrm.org/dev/core/issues/392). It is also worth knowing that both [Backdrop](https://forum.backdropcms.org/forum/installing-backdrop-1126-mysql-8-sqlmode-cant-be-set-value-noautocreateuser) and [Drupal 7](https://www.drupal.org/project/drupal/issues/2978575) have open issues with regards to MySQL 8 support. [Drupal 8](https://www.drupal.org/docs/8/system-requirements/database-server) supports MySQL 8 as of version 8.6 as per [this issue](https://www.drupal.org/project/drupal/issues/2966523), Current versions of WordPress and [Joomla](https://docs.joomla.org/Joomla_and_MySQL_8) appear to be compatible with MySQL 8
```
Comment (eileenmcnaughton) [inline docs/requirements.md:112]: Can we change "It is not yet recommended as" to just "However,"

File: docs/requirements.md (hunk 1 above)
Comment (MikeyMJCO) [inline docs/requirements.md:108]:
```suggestion
Your MySQL version should be **5.7.5 or greater** or MariaDB **10.0.2 or greater**.  As of version 5.26 the minimum install version for CiviCRM is 5.5, users on versions before that are advised to upgrade their MySQL instance to a recommended version. CiviCRM may not run correctly on MySQL 5.5 and/or 5.6 as these versions don't support some of the functionality CiviCRM uses to ensure that database actions don't compete for the same resources.
```

File: docs/requirements.md (hunk 2 above)
Comment (MikeyMJCO) [inline docs/requirements.md:112]:
```suggestion
As of version 5.24 CiviCRM has been shown to be able to run on MySQL 8 through the execution of our test matrix. Not all of the Content Management Systems support MySQL 8, CiviCRM MySQL 8 support is being tracked in this [open issue for MySQL 8 support](https://lab.civicrm.org/dev/core/issues/392). It is also worth knowing that both [Backdrop](https://forum.backdropcms.org/forum/installing-backdrop-1126-mysql-8-sqlmode-cant-be-set-value-noautocreateuser) and [Drupal 7](https://www.drupal.org/project/drupal/issues/2978575) have open issues with regards to MySQL 8 support. [Drupal 8](https://www.drupal.org/docs/8/system-requirements/database-server) [supports MySQL 8 as of version 8.6](https://www.drupal.org/project/drupal/issues/2966523), Current versions of WordPress and [Joomla](https://docs.joomla.org/Joomla_and_MySQL_8) appear to be compatible with MySQL 8
```

**Comment by MikeyMJCO** [comment]
@eileenmcnaughton can you re-review - I actioned your suggestions :-)

**Comment by eileenmcnaughton** [comment]
@MikeyMJCO looks good to me

**Pull request merged**